### DISCUSSION TITLE
Trying to run LAMP server on SO box

### DISCUSSION THREAD

**Discussion by gigantor2** [Security-Onion-Solutions/securityonion#4642] [Unsupported Versions] [answered]
After further study I am back to my initial theory of trying to run the  SO Nginx server as a Reverse Proxy. I would like Nginx to be a front end server and forward all requests that come to 80/443 to my Lamp web server. Right now, any requests to the host on 80/443 are sent to the 80/443 Nginx container SO sign-in page. This is not good!!! 

I currently have SO 2.3.50 installed and running on Docker in a  centos server ("stand alone" server installed by ISO) I also have a LAMP web server installed (which was installed before SO). After the SO install all web requests went through to SO and would no longer get through to my LAMP server. Essentially, SO took over the host and is not letting requests through to my LAMP Containers on any port. My Lamp web server's port 80 and 443 are mapped to other ports on the host via my compose file, so even though SO owns port 80 and 443 on the host, I thought I would be able to get through on the other mapped ports. I have tried all kinds of schemes to get this to work ( putting mgmt port on second interface, and have the primary interface for LAMP, Having 3 interfaces. and allowing LAMP be primary) , but to no avail. I believe I need Nginx to forward those requests, or some kind of pass through to my LAMP server.

BTW- I have no errors, the lamp server is up, listening, and ready to accept requests like it was prior to the S.O. install. In addition, S.O. is accessible via my browser and appears that all applications I selected are available and working. However, I cannot access web pages via the browser without attaching the other mapped ports to the end of the URL. (192.168.1.100:8330) I can then access web pages, but that will not work in a operational environment. Persons accessing my site won't know to include a unique port.

My goal is to have security around my web server in a Docker environment. They both work well separately, but I need them to work together. I read something in my research about SO allowing apps to "PASS TROUGH," but after logging into the SO Nginx container  it appeared the code was just for the SO applications (Kibana, Wazuh, etc.). Am I wrong, can the code be added for my LAMP apps.

BTW-  I am happy SO is working so well and not allowing anything through to the server. This is exactly what I want it to do, protect my web server. I just want it block everything going anywhere else but the web server ( port 80 or 443). I don't want to have to have two servers at this point ( LAMP and SO, maybe at a later date), I would like them to work together in tandem on the same box. My reason for this is the cost of running two servers on a hosting service. 

Am I wasting my time and spinning my wheels, or can this be done by a novice like myself? Any help would thoroughly  appreciated.

**Comment by weslambert** [comment, marked as answer]
While it is possible, we don't support running Security Onion alongside other components, unless specifically noted as compatible.  There are many different variables, depending on exactly how things are configured. This is simply not something for which we can provide free support or guarantee will work.

**Comment by gigantor2** [comment]
Thank you for your reply. I am just checking my email since I made those post previouly regarding running LAMP with SO. I didn't realize your replies would be coming to my email, so I was checking where i made the original post for a reply. I just thought you guys were too busy for my question. I am pleasantly pleased to know that you saw my question and at least replied to let me know that my configuration was not supported.

The good thing about me digging in and trying to make this work is that I found out how good SO is at stopping other applications from running on the host on which it occupies. SO prevented me from loading backup jpegs into the LAMP web server, and would not let me serve any pages via 443 or 80 on my Apache2 server. So from a security standpoint it seems to take over the host and stop certain components of the LAMPapplication from functioning even though I was able to install it on the host.

As a result, I came to the conclusion that I will have to run SO and my LAMP server on separate hosts or VMs. i also learned how difficult it is to use the "pass through" function in the nginx container to pass outside applications through any ports at all, must less, ports 443 or 80. Seems to me it is "locked down" pretty well in the "host groups" and "port groups" from the nginx docker image and it future containers that are created.

Congratulations on a great application, you guys did a great job, especially including Elastic Stack functionality in SO.  Everything is currently working great when I use them individually ("Knock on wood")Thanks again for responding to my question, I take it as a great learning experience.

Herbert E. Gregory Senior Security Engineer/Consultant ??CISA, MCSE, CCNA, MCP, MPA
http://www.linkedin.com/in/hgregory
________________________________
